pkg/config/validation/validation.go

```diff
@@ -2779,15 +2779,26 @@ func validateHTTPRetry(retries *networking.HTTPRetry) (errs error) {
 }
 
 func validateHTTPRedirect(redirect *networking.HTTPRedirect) error {
-	if redirect != nil && redirect.Uri == "" && redirect.Authority == "" {
-		return errors.New("redirect must specify URI, authority, or both")
+	if redirect == nil {
+		return nil
+	}
+	if redirect.Uri == "" && redirect.Authority == "" && redirect.RedirectPort == nil && redirect.Scheme == "" {
+		return errors.New("redirect must specify URI, authority, scheme, or port")
 	}
 
-	if redirect != nil && redirect.RedirectCode != 0 {
+	if redirect.RedirectCode != 0 {
 		if redirect.RedirectCode < 300 || redirect.RedirectCode > 399 {
 			return fmt.Errorf("%d is not a valid redirect code, must be 3xx", redirect.RedirectCode)
 		}
 	}
+	if redirect.Scheme != "" && redirect.Scheme != "http" && redirect.Scheme != "https" {
```

Comment:
but in  gateway conversion, it is `HTTP` or `HTTPS`

Reply:
We do strings.ToLower there though

Reply:
make sense, i missed it